lib/kafka/protocol/message_set.rb

```diff
@@ -3,9 +3,14 @@ module Protocol
     class MessageSet
       attr_reader :messages
 
-      def initialize(messages: [], compression_codec: nil)
+      def initialize(messages: [], compression_codec: nil, compression_threshold: 1)
```

Comment:
I'm not quite sure I understand your feedback – I assume you want to have a null codec that simply doesn't touch the messages?

The problem with that is that the code paths for when compression is enabled and when it's disabled aren't the same – there's an additional "wrap" step we have to go through when compressing.